src/backend/storage/file/buffile.c

```diff
@@ -1012,6 +1042,12 @@ BufFileStartCompression(BufFile *file)
 {
 	ResourceOwner oldowner;
 	size_t ret;
+	ZSTD_customMem customMem;
+	ResourceOwnerContext resContext;
+
+	customMem.customAlloc = customAlloc;
+	customMem.customFree = customFree;
+	customMem.opaque = &resContext;
```

Comment:
Is opaque only used while the BufFileStartCompression function is running?

Reply:
Now we assign file->resowner to opaque which is a regular pointer